### ISSUE TITLE
Geocoder non-gated CORS Support 

### ISSUE THREAD

**Issue by xemoka** [bcgov/api-specs#133]
Hello, 

I'm attempting to use the Geocorder Rest API for geocoding in a web application we're (Selkirk College) developing in collaboration with an industry partner and I'm running into a small problem with your geocoding API (which otherwise is quite excellent), it appears as though you do not provide CORS headers (Access-Control-Allow-Origin)—I feel this could be an oversight as the documentation and [previous release notes](http://www2.gov.bc.ca/gov/content/governments/about-the-bc-government/databc/geographic-data-and-services/geocoder/whats-new-geocoder) (May 12, 2014) seems to indicate that you are interested in others using this API, however—as I'm sure you are aware—without proper CORS headers modern browsers reject the request. 

Is this an oversight or are you wanting users to provide a proxy on their own servers to prevent CSRF? 

Thanks for your time.

**Comment by cmhodgson** [comment]
I'm not sure when this got dropped, it was unintentional. I'm surprised it hasn't affected other users, in fact, I'm not sure I understand how some of my demo apps are working without this... Leo, is Kong doing something for CORS? I think the problem is that the java Spring code is bypassing the CorsFilter I had previously setup at the tomcat level. Apparently I just need to an "@CrossOrigin" annotation to make this work in Spring.

**Comment by ll911** [comment]
cors is handled by the gateway, non-gated geocoder cors is only enabled for .gov.bc.ca domain.
also works for website handles csp headers.

Sent from Mobile




On Mon, Jan 30, 2017 at 10:49 AM -0800, "Chris Hodgson" <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm not sure when this got dropped, it was unintentional. I'm surprised it hasn't affected other users, in fact, I'm not sure I understand how some of my demo apps are working without this... Leo, is Kong doing something for CORS? I think the problem is that the java Spring code is bypassing the CorsFilter I had previously setup at the tomcat level. Apparently I just need to an "@CrossOrigin<https://github.com/CrossOrigin>" annotation to make this work in Spring.

**Comment by xemoka** [comment]
I've been in contact with @mraross via email, he's confirmed this is intentional—I'm awaiting an API key(s?) for the gated version and will most likely continue to use it. 

It would be nice if the documentation was to mention the requirement for an API key / gated access (although, perhaps I missed it?); I'm happily using it now (time invested and good results for my purposes), however, probably I would have rolled my own / found alternatives if it was more clear, due to time and project constraints—initially I just figured it was due to localhost development that I was having to force cross-origin headers on the response.

Thanks again.
